tensorflow/compiler/jit/deadness_analysis.cc

```diff
@@ -1170,22 +1462,14 @@ DeadnessAnalysisImpl::PredicateMapAsString() const {
 }
 
 namespace deadness_analysis_internal {
-Status ComputePredicates(const Graph& graph,
-                         PredicateMapTy* out_predicate_map) {
+Status ComputePredicates(const Graph& graph, PredicateMapTy* out_predicate_map,
+                         bool force_pessimistic) {
   DeadnessAnalysisImpl impl(&graph);
-  TF_RETURN_IF_ERROR(impl.Populate());
+  TF_RETURN_IF_ERROR(impl.Populate(force_pessimistic));
```

Comment:
Is there value in testing with `force_pessimistic=true`?  I'd venture that we should test the default mode of analysis unless there is reason to do otherwise.

Reply:
The unittest tests both cases if the results are different. On the other hand, the optimistic mode is tested when fallback is expected.